Mage/src/main/java/mage/abilities/effects/common/counter/AddCountersSourceEffect.java

```diff
@@ -70,61 +70,67 @@ public AddCountersSourceEffect(final AddCountersSourceEffect effect) {
     @Override
     public boolean apply(Game game, Ability source) {
         Player controller = game.getPlayer(source.getControllerId());
-        if (controller != null) {
-            if (putOnCard) {
-                Card card = game.getCard(source.getSourceId());
-                if (card != null) {
-                    if (counter != null) {
-                        Counter newCounter = counter.copy();
-                        int countersToAdd = amount.calculate(game, source, this);
-                        if (countersToAdd > 0 && newCounter.getCount() == 1) {
-                            countersToAdd--;
-                        }
-                        newCounter.add(countersToAdd);
-                        List<UUID> appliedEffects = (ArrayList<UUID>) this.getValue("appliedEffects");
-                        card.addCounters(newCounter, source.getControllerId(), source, game, appliedEffects);
-                        if (informPlayers && !game.isSimulation()) {
-                            Player player = game.getPlayer(source.getControllerId());
-                            if (player != null) {
-                                game.informPlayers(player.getLogName() + " puts " + newCounter.getCount() + ' ' + newCounter.getName().toLowerCase(Locale.ENGLISH) + " counter on " + card.getLogName());
-                            }
-                        }
-                    }
-                    return true;
-                }
-            } else {
-                Permanent permanent = game.getPermanent(source.getSourceId());
-                if (permanent == null && source.getAbilityType() == AbilityType.STATIC) {
-                    permanent = game.getPermanentEntering(source.getSourceId());
+        if (controller == null) {
+            return false;
+        }
+
+        if (counter == null) {
```

Comment:
Small nitpick: I would combine this check with the above into one check

Reply:
I actually think I should put the counter one at the very top, since if that's null then there's no point to even try to look at the player.